src/snmalloc/backend/backend.h

```diff
@@ -1,223 +1,27 @@
 #pragma once
 #include "../backend_helpers/backend_helpers.h"
 
-#if defined(SNMALLOC_CHECK_CLIENT) && !defined(OPEN_ENCLAVE)
-/**
- * Protect meta data blocks by allocating separate from chunks for
- * user allocations. This involves leaving gaps in address space.
- * This is less efficient, so should only be applied for the checked
- * build.
- *
- * On Open Enclave the address space is limited, so we disable this
- * feature.
- */
-#  define SNMALLOC_META_PROTECTED
-#endif
-
 namespace snmalloc
 {
   /**
    * This class implements the standard backend for handling allocations.
    * It abstracts page table management and address space management.
    */
-  template<SNMALLOC_CONCEPT(ConceptPAL) PAL, bool fixed_range>
-  class BackendAllocator : public CommonConfig
+  template<
+    SNMALLOC_CONCEPT(ConceptPAL) PAL,
+    typename PageMapEntry,
```

Comment:
Does this alias with the PageMapEntry class name? Would it make sense to have a slight variation of the name just to avoid confusion?

Reply:
Made the class `DefaultPagemapEntry`, so it is clear when it is used to instantiate something `PagemapEntry`.